src/Domain/EmailAddress.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Embeddable;
+use MatchBot\Application\Assertion;
+
+/** @Embeddable */
```

Comment:
slight issue with using an Embeddable value objects - in some cases the ORM can change the value even if we thought they were immutable, see https://github.com/doctrine/orm/issues/7539 . That would be a problem if the VO was shared between two things, which is fine for a genuinely immutable object. But that's only an issue when use `EntityManager#refresh`, which we don't use much. 

Or we only use for Donations: https://github.com/thebiggive/matchbot/blob/bd90fab20a39b9f6a00dd5f5e0709e586274067a/src/Domain/DonationRepository.php#L698